packages/workspace/src/browser/workspace-service.ts

```diff
@@ -88,7 +92,10 @@ export class WorkspaceService implements FrontendApplicationContribution {
         const valid = await this.toValidRoot(rootUri);
         if (valid) {
             // The same window has to be preserved too (instead of opening a new one), if the workspace root is not yet available and we are setting it for the first time.
-            const preserveWindow = options ? options.preserveWindow : !(await this.root);
```

Comment:
How about:
```ts
const { preserveWindow } = {
    preserveWindow: this.preferences['workspace.preserveWindow'] || !(await this.root),
    ...options
};
```

It is the pattern which i find useful to define defaults and then override them by arguments if they are provided. After https://github.com/theia-ide/theia/pull/1709#issuecomment-382531492 I've realized that the PR already does what i want but it was not clear for me when i had read the code first time.

Reply:
This is much cleaner and easy to understand thanks!